src/shared/nested-dropdown-list/list-item.tsx

```diff
@@ -67,7 +81,13 @@ export const ListItem = <V1, V2, V3>({
     // HELPER FUNCTIONS
     // =============================================================================
     const checkListItemSelected = (keyPath: string[]): boolean => {
-        return JSON.stringify(selectedKeyPath) === JSON.stringify(keyPath);
+        if (!multiSelect) {
+            return JSON.stringify(selectedKeyPath) === JSON.stringify(keyPath);
+        } else {
+            return selectedKeyPaths.some(
+                (key) => JSON.stringify(key) === JSON.stringify(keyPath)
```

Comment:
actually can the `item.checked` state be used here instead

Reply:
due to the changes for `selectedKeyPaths: string[][]` now. below fn can use for single and nested. let me know if still need the `item.checked`

```
 const checkListItemSelected = (keyPath: string[]): boolean =>
        selectedKeyPaths.some(
            (key) => JSON.stringify(key) === JSON.stringify(keyPath)
        );
```